react-components/src/ToolDetails.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @author psarando
+ */
+import React, { Component } from 'react';
+
+class ToolDetails extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedToolIndex: 0
+        };
+
+        // This binding is necessary to make `this` work in the callback
+        this.onToolSelectionChange = this.onToolSelectionChange.bind(this);
+    }
+
+    onToolSelectionChange(event) {
+        let selectedToolIndex = event.target.value;
+
+        this.setState({
+            selectedToolIndex: selectedToolIndex
+        });
+    }
+
+    render() {
+        let tools      = this.props.app.tools,
+            appearance = this.props.appearance,
+            toolInfo   = tools[this.state.selectedToolIndex],
+            labelClass = appearance.css().label(),
+            valueClass = appearance.css().value();
+
+        return (
+            <div>
+                <select value={this.state.selectedToolIndex}
+                        onChange={this.onToolSelectionChange} >
+                    {
+                        tools.map( (toolInfo, index) => (
+                                <option key={index} value={index}>
+                                    {toolInfo.name}
+                                </option>
+                            )
+                        )
+                    }
+                </select>
+                <table>
+                    <tbody>
+                    <tr>
+                        <td>
+                            {appearance.detailsLabel()}
+                        </td>
+                        <td>
+                            <i>Rendered by React!</i>
+                        </td>
```

Comment:
Compare the JSX in this file with [ToolDetailsView.ui.xml](https://github.com/psarando/cyverse-de-ui/blob/master/de-lib/src/main/java/org/iplantc/de/apps/client/views/details/ToolDetailsView.ui.xml).